src/main/java/com/aws/greengrass/security/SecurityService.java

```diff
@@ -163,6 +165,47 @@ public KeyManager[] getKeyManagers(URI privateKeyUri, URI certificateUri)
         return provider.getKeyManagers(privateKeyUri, certificateUri);
     }
 
+    /**
+     * Get certificate chain using private and certificate URIs.
+     *
+     * @param privateKeyUri  private key URI
+     * @param certificateUri certificate URI
+     * @return X509Certificate list that corresponds to the given private key and certificate URIs
+     * @throws ServiceUnavailableException if crypto key provider service is unavailable
+     * @throws KeyLoadingException         if crypto key provider service fails to load key
+     */
+    public X509Certificate[] getCertificateChain(URI privateKeyUri, URI certificateUri)
+            throws ServiceUnavailableException, KeyLoadingException {
+        logger.atTrace().kv(KEY_URI, privateKeyUri).kv(CERT_URI, certificateUri)
+                .log("Getting the certificate chain using private key and certificate URIs");
+
+        X509Certificate[] certChain = new X509Certificate[0];
+        CryptoKeySpi provider = selectCryptoKeyProvider(privateKeyUri);
+        KeyManager[] km = provider.getKeyManagers(privateKeyUri, certificateUri);
+
+        if (km.length == 1 && km[0] instanceof X509KeyManager) {
+            X509KeyManager x509KeyManager = (X509KeyManager) km[0];
+            KeyPair keyPair = provider.getKeyPair(privateKeyUri, certificateUri);
+            String[] aliases = x509KeyManager.getClientAliases(keyPair.getPublic().getAlgorithm(), null);
+            logger.atTrace().kv(KEY_URI, privateKeyUri).kv(CERT_URI, certificateUri)
+                    .log("Getting the certificate chain from the provider using the key manager");
+            if (aliases == null) {
+                logger.atError().kv(KEY_URI, privateKeyUri).kv(CERT_URI, certificateUri)
+                        .log("Unable to find aliases in the key manager.");
+                return certChain;
```

Comment:
this is an error case, shouldn't we be throwing an exception, not an empty array?

Reply:
I think it's on the caller to decide what to do with the output. It might not always be an exception on the other side.

Reply:
With this pattern though, the caller doesn't really know that anything has gone wrong. It isn't a pattern that everyone always agrees with, but Greengrass code guidelines do have us used checked exceptions. So use a checked exception for exceptional cases (which this is).